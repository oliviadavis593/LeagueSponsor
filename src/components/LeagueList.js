import React, { Component } from 'react';
import LeagueItem from '../components/LeagueItem';
import LeagueContext from '../LeagueContext';
import '../styles/LeagueList.css';

class LeagueList extends Component {

    static contextType = LeagueContext; 

    render() {
        const leagues = this.context.leagues; 
        return(
            <div>
                <div className='ls-dashboard__title'>
                    <h1>All Leagues</h1>
                </div>
                <section className='ls-league-list__container'>
                <ul className='ls-league_list'>
                    {leagues.map(league => (
                        <li key={league.id}>
                            <LeagueItem 
                            id={league.id}
                            league_name={league.league_name}
                            website={league.website}
                            location={league.location}
                            price={league.price}
                            latitude={league.latitude}
                            longitude={league.longitude}
                            />
                        </li>
                    ))}
                </ul>
            </section>
            </div>
        )
    }
}

export default LeagueList; 