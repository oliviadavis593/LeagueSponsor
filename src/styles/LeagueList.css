.ls-league_list {
    list-style-type: none; 
    text-align: center; 
    margin: 30px auto; 
    width: 250px;   
}

.ls-dashboard__title {
    margin-top: 50px;
    font-size: 25px;  
}


