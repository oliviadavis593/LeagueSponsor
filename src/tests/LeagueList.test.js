import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import LeagueList from '../components/LeagueList';

describe('LeagueList Component', () => {
    it(`renders without crashing`, () => {
        const div = document.createElement('div')
        ReactDOM.render(<BrowserRouter><LeagueList /></BrowserRouter>,div)
        ReactDOM.unmountComponentAtNode(div)
    })

    it('renders the UI as expected', () => {
        const tree = renderer 
            .create(<BrowserRouter><LeagueList /></BrowserRouter>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    })
})