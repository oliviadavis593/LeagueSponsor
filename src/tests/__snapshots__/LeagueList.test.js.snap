// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeagueList Component renders the UI as expected 1`] = `
<div>
  <div
    className="ls-dashboard__title"
  >
    <h1>
      All Leagues
    </h1>
  </div>
  <section
    className="ls-league-list__container"
  >
    <ul
      className="ls-league_list"
    >
      <li>
        <div
          className="ls-league"
        >
          <h2
            className="ls-league__name"
          >
            <a
              href="https://www.mlb.com/cardinals"
              target="blank"
            >
              St. Louis Cardinals
            </a>
          </h2>
          <div
            className="ls-league__location"
          >
            Location: 
            St. Louis, Missouri
          </div>
          <div
            className="ls-league__price"
          >
            Price: 
            $4,500
          </div>
        </div>
      </li>
      <li>
        <div
          className="ls-league"
        >
          <h2
            className="ls-league__name"
          >
            <a
              href="https://www.mlb.com/braves"
              target="blank"
            >
              Atlanta Braves
            </a>
          </h2>
          <div
            className="ls-league__location"
          >
            Location: 
            Atlanta, Georgia
          </div>
          <div
            className="ls-league__price"
          >
            Price: 
            $1,400
          </div>
        </div>
      </li>
      <li>
        <div
          className="ls-league"
        >
          <h2
            className="ls-league__name"
          >
            <a
              href="https://www.mlb.com/marlins"
              target="blank"
            >
              Miami Marlins
            </a>
          </h2>
          <div
            className="ls-league__location"
          >
            Location: 
            Miami, Florida
          </div>
          <div
            className="ls-league__price"
          >
            Price: 
            $800
          </div>
        </div>
      </li>
      <li>
        <div
          className="ls-league"
        >
          <h2
            className="ls-league__name"
          >
            <a
              href="https://www.mlb.com/mets"
              target="blank"
            >
              New York Mets
            </a>
          </h2>
          <div
            className="ls-league__location"
          >
            Location: 
            Queens, New York, New York
          </div>
          <div
            className="ls-league__price"
          >
            Price: 
            $9,500
          </div>
        </div>
      </li>
      <li>
        <div
          className="ls-league"
        >
          <h2
            className="ls-league__name"
          >
            <a
              href="https://www.mlb.com/phillies"
              target="blank"
            >
              Philadelphia Phillies
            </a>
          </h2>
          <div
            className="ls-league__location"
          >
            Location: 
            Philadelphia, PA
          </div>
          <div
            className="ls-league__price"
          >
            Price: 
            $3,500
          </div>
        </div>
      </li>
      <li>
        <div
          className="ls-league"
        >
          <h2
            className="ls-league__name"
          >
            <a
              href="https://www.mlb.com/nationals"
              target="blank"
            >
              Washington Nationals
            </a>
          </h2>
          <div
            className="ls-league__location"
          >
            Location: 
            Washington, D.C.
          </div>
          <div
            className="ls-league__price"
          >
            Price: 
            $7,500
          </div>
        </div>
      </li>
      <li>
        <div
          className="ls-league"
        >
          <h2
            className="ls-league__name"
          >
            <a
              href="https://www.mlb.com/cubs"
              target="blank"
            >
              Chicago Cubs
            </a>
          </h2>
          <div
            className="ls-league__location"
          >
            Location: 
            Chicago, Illinois
          </div>
          <div
            className="ls-league__price"
          >
            Price: 
            $10,000
          </div>
        </div>
      </li>
      <li>
        <div
          className="ls-league"
        >
          <h2
            className="ls-league__name"
          >
            <a
              href="https://www.mlb.com/reds"
              target="blank"
            >
              Cincinnati Reds
            </a>
          </h2>
          <div
            className="ls-league__location"
          >
            Location: 
            Cincinnati, Ohio
          </div>
          <div
            className="ls-league__price"
          >
            Price: 
            $400
          </div>
        </div>
      </li>
      <li>
        <div
          className="ls-league"
        >
          <h2
            className="ls-league__name"
          >
            <a
              href="https://www.mlb.com/brewers"
              target="blank"
            >
              Milwaukee Brewers
            </a>
          </h2>
          <div
            className="ls-league__location"
          >
            Location: 
            Milwaukee, Wisconsin
          </div>
          <div
            className="ls-league__price"
          >
            Price: 
            $2,250
          </div>
        </div>
      </li>
      <li>
        <div
          className="ls-league"
        >
          <h2
            className="ls-league__name"
          >
            <a
              href="https://www.mlb.com/pirates"
              target="blank"
            >
              Pittsburgh Pirates
            </a>
          </h2>
          <div
            className="ls-league__location"
          >
            Location: 
            Pittsburgh, Pennsylvania
          </div>
          <div
            className="ls-league__price"
          >
            Price: 
            $3,000
          </div>
        </div>
      </li>
      <li>
        <div
          className="ls-league"
        >
          <h2
            className="ls-league__name"
          >
            <a
              href="https://www.mlb.com/dbacks"
              target="blank"
            >
              Arizona Diamondbacks
            </a>
          </h2>
          <div
            className="ls-league__location"
          >
            Location: 
            Arizona
          </div>
          <div
            className="ls-league__price"
          >
            Price: 
            $1,000
          </div>
        </div>
      </li>
      <li>
        <div
          className="ls-league"
        >
          <h2
            className="ls-league__name"
          >
            <a
              href="https://www.mlb.com/rockies"
              target="blank"
            >
              Colorado Rockies
            </a>
          </h2>
          <div
            className="ls-league__location"
          >
            Location: 
            Denver, Colorado
          </div>
          <div
            className="ls-league__price"
          >
            Price: 
            $5,550
          </div>
        </div>
      </li>
      <li>
        <div
          className="ls-league"
        >
          <h2
            className="ls-league__name"
          >
            <a
              href="https://www.mlb.com/dodgers"
              target="blank"
            >
              Los Angeles Dodgers
            </a>
          </h2>
          <div
            className="ls-league__location"
          >
            Location: 
            Los Angeles, California
          </div>
          <div
            className="ls-league__price"
          >
            Price: 
            $14,500
          </div>
        </div>
      </li>
      <li>
        <div
          className="ls-league"
        >
          <h2
            className="ls-league__name"
          >
            <a
              href="https://www.mlb.com/padres"
              target="blank"
            >
              San Diego Padres
            </a>
          </h2>
          <div
            className="ls-league__location"
          >
            Location: 
            San Diego, California
          </div>
          <div
            className="ls-league__price"
          >
            Price: 
            $5,700
          </div>
        </div>
      </li>
      <li>
        <div
          className="ls-league"
        >
          <h2
            className="ls-league__name"
          >
            <a
              href="https://www.mlb.com/giants"
              target="blank"
            >
              San Francisco Giants
            </a>
          </h2>
          <div
            className="ls-league__location"
          >
            Location: 
            San Francisco, California
          </div>
          <div
            className="ls-league__price"
          >
            Price: 
            $1,200
          </div>
        </div>
      </li>
      <li>
        <div
          className="ls-league"
        >
          <h2
            className="ls-league__name"
          >
            <a
              href="https://www.mlb.com/orioles"
              target="blank"
            >
              Baltimore Orioles
            </a>
          </h2>
          <div
            className="ls-league__location"
          >
            Location: 
            Baltimore, Maryland
          </div>
          <div
            className="ls-league__price"
          >
            Price: 
            $2,200
          </div>
        </div>
      </li>
      <li>
        <div
          className="ls-league"
        >
          <h2
            className="ls-league__name"
          >
            <a
              href="https://www.mlb.com/indians"
              target="blank"
            >
              Cleveland Indians
            </a>
          </h2>
          <div
            className="ls-league__location"
          >
            Location: 
            Clevland, Ohio
          </div>
          <div
            className="ls-league__price"
          >
            Price: 
            $1,400
          </div>
        </div>
      </li>
      <li>
        <div
          className="ls-league"
        >
          <h2
            className="ls-league__name"
          >
            <a
              href="https://www.mlb.com/redsox"
              target="blank"
            >
              Boston Red Sox
            </a>
          </h2>
          <div
            className="ls-league__location"
          >
            Location: 
            Boston, Massachusetts
          </div>
          <div
            className="ls-league__price"
          >
            Price: 
            $15,000
          </div>
        </div>
      </li>
      <li>
        <div
          className="ls-league"
        >
          <h2
            className="ls-league__name"
          >
            <a
              href="https://www.mlb.com/yankees"
              target="blank"
            >
              New York Yankees
            </a>
          </h2>
          <div
            className="ls-league__location"
          >
            Location: 
            New York, New York
          </div>
          <div
            className="ls-league__price"
          >
            Price: 
            $8,600
          </div>
        </div>
      </li>
      <li>
        <div
          className="ls-league"
        >
          <h2
            className="ls-league__name"
          >
            <a
              href="https://www.mlb.com/rays"
              target="blank"
            >
              Tampa Bay Rays
            </a>
          </h2>
          <div
            className="ls-league__location"
          >
            Location: 
            Tampa Bay, Florida
          </div>
          <div
            className="ls-league__price"
          >
            Price: 
            $1,300
          </div>
        </div>
      </li>
    </ul>
  </section>
</div>
`;
